python/dgllife/data/pdbbind.py

```diff
@@ -73,34 +78,132 @@ class PDBBind(object):
+        elif pdb_version == 'v2015':
+            self._url = 'dataset/pdbbind_v2015.tar.gz'
+            data_path = root_dir_path + '/pdbbind_v2015.tar.gz'
+            extracted_data_path = root_dir_path + '/pdbbind_v2015'
+            download(_get_dgl_url(self._url), path=data_path, overwrite=False)
+            extract_archive(data_path, extracted_data_path)
+
+            if subset == 'core':
+                index_label_file = extracted_data_path + '/v2015/INDEX_core_data.2013'
+            elif subset == 'refined':
+                index_label_file = extracted_data_path + '/v2015/INDEX_refined_data.2015'
+            else:
+                raise ValueError(
+                    'Expect the subset_choice to be either '
                 'core or refined, got {}'.format(subset))
+        elif pdb_version == 'v2007':
+            # download from pddbind official website
+        #     download_url = 'http://www.pdbbind.org.cn/download/pdbbind_v2007.tar.gz' 
+        #     data_path = root_dir_path + '/pdbbind_v2007.tar.gz'
+        #     extracted_data_path = root_dir_path + '/pdbbind_v2007'
+        #     download(download_url, path=data_path, overwrite=False)
+        #     extract_archive(data_path, extracted_data_path)
+            self._url = 'dataset/pdbbind_v2007.tar.gz'
+            data_path = root_dir_path + '/pdbbind_v2007.tar.gz'
+            extracted_data_path = root_dir_path + '/pdbbind_v2007'
+            download(_get_dgl_url(self._url), path=data_path, overwrite=False)
+            extract_archive(data_path, extracted_data_path, overwrite=False)
+            extracted_data_path += '/home/ubuntu' # extra layer 
 
-        self._preprocess(extracted_data_path, index_label_file, load_binding_pocket,
-                         sanitize, calc_charges, remove_hs, use_conformation,
-                         construct_graph_and_featurize, zero_padding, num_processes)
+            # DataFrame containing the pdbbind_2007_agglomerative_split.txt
+            self.agg_split = pd.read_csv(extracted_data_path + '/v2007/pdbbind_2007_agglomerative_split.txt')
+            self.agg_split.rename(columns={'PDB ID':'PDB_code', 'Sequence-based assignment':'sequence', 'Structure-based assignment':'structure'}, inplace=True)
+            self.agg_split.loc[self.agg_split['PDB_code']=='1.00E+66', 'PDB_code'] = '1e66' # fix typo
+            if subset == 'core':
+                index_label_file = extracted_data_path + '/v2007/INDEX.2007.core.data'
+            elif subset == 'refined':
+                index_label_file = extracted_data_path + '/v2007/INDEX.2007.refined.data'
+            else:
+                raise ValueError(
+                    'Expect the subset_choice to be either '
+                'core or refined, got {}'.format(subset))
+
+        contents = []
+        with open(index_label_file, 'r') as f:
+            for line in f.readlines():
+                if line[0] != "#":
+                    splitted_elements = line.split()
```

Comment:
How should we handle custom data here?

Reply:
User should leave pdbbind_version to default v2015, and make sure the local file formats are consistent with PDBBind v2015